datumaro/datumaro/plugins/yolo_format/extractor.py

```diff
@@ -134,21 +127,30 @@ def _get(self, item_id, subset_name):
         return item
 
     @staticmethod
-    def _parse_annotations(anno_path, image_width, image_height):
+    def _parse_annotations(anno_path, image):
+        lines = []
         with open(anno_path, 'r') as f:
-            annotations = []
             for line in f:
-                label_id, xc, yc, w, h = line.strip().split()
-                label_id = int(label_id)
-                w = float(w)
-                h = float(h)
-                x = float(xc) - w * 0.5
-                y = float(yc) - h * 0.5
-                annotations.append(Bbox(
-                    round(x * image_width, 1), round(y * image_height, 1),
-                    round(w * image_width, 1), round(h * image_height, 1),
-                    label=label_id
-                ))
+                line = line.strip()
```

Comment:
@zhiltsov-max , I don't think that it is a good idea to read the whole file into memory. I understand that the code below does it in any case but probably annotations.append can be clever in the future.

Reply:
These files contain annotations only for a single image, so it's unlikely to become a bottleneck. Whole dataset jsons in COCO hurt much more.